drivers/i2s/i2s_ll_stm32.c

```diff
@@ -0,0 +1,1171 @@
+	},
+};
+DEVICE_AND_API_INIT(i2s_stm32_4, CONFIG_I2S_4_NAME, &i2s_stm32_initialize,
+		    &i2s_stm32_data_4, &i2s_stm32_config_4, POST_KERNEL,
+		    CONFIG_I2S_INIT_PRIORITY, &i2s_stm32_driver_api);
+
+static void i2s_stm32_irq_config_func_4(struct device *dev)
+{
+	SYS_LOG_DBG("Enable I2S4 IRQ");
+	IRQ_CONNECT(CONFIG_SPI_4_IRQ, CONFIG_SPI_4_IRQ_PRI, i2s_stm32_isr,
+		    DEVICE_GET(i2s_stm32_4), 0);
+	irq_enable(CONFIG_SPI_4_IRQ);
+}
+
+#endif /* CONFIG_I2S_4 */
+
+#ifdef CONFIG_I2S_5
+static struct device DEVICE_NAME_GET(i2s_stm32_5);
+
+static void i2s_stm32_irq_config_func_5(struct device *dev);
+
+static const struct i2s_stm32_cfg i2s_stm32_config_5 = {
+	.i2s = (SPI_TypeDef *) CONFIG_SPI_5_BASE_ADDRESS,
+	.pclken = {
+		.enr = LL_APB2_GRP1_PERIPH_SPI5,
+		.bus = STM32_CLOCK_BUS_APB2,
+	},
+#ifdef CONFIG_I2S_STM32_USE_PLLI2S_ENABLE
+	.i2s_clk_sel = LL_RCC_I2S2_CLKSOURCE_PLLI2S,
+#else
+	.i2s_clk_sel = LL_RCC_I2S2_CLKSOURCE_PLLSRC,
+#endif
+	.irq_config = i2s_stm32_irq_config_func_5,
+};
+
+struct queue_item rx_5_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
+struct queue_item tx_5_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
+
+static struct i2s_stm32_data i2s_stm32_data_5 = {
+	.dma_name = I2S5_DMA_NAME,
+	.rx = {
+		.dma_channel = I2S5_DMA_CHAN_RX,
+		.dma_cfg = {
+			.block_count = 1,
+			.dma_slot = I2S5_DMA_SLOT_RX,
+			.channel_direction = PERIPHERAL_TO_MEMORY,
+			.source_data_size = 1,  /* 16bit default */
+			.dest_data_size = 1,    /* 16bit default */
+			.source_burst_length = 0, /* SINGLE transfer */
+			.dest_burst_length = 1,
+			.dma_callback = dma_rx_callback,
+		},
+		.stream_start = rx_stream_start,
+		.stream_disable = rx_stream_disable,
+		.queue_drop = rx_queue_drop,
+		.mem_block_queue.buf = rx_5_ring_buf,
+		.mem_block_queue.len = ARRAY_SIZE(rx_5_ring_buf),
+	},
+	.tx = {
+		.dma_channel = I2S5_DMA_CHAN_TX,
+		.dma_cfg = {
+			.block_count = 1,
+			.dma_slot = I2S5_DMA_SLOT_TX,
+			.channel_direction = MEMORY_TO_PERIPHERAL,
+			.source_data_size = 1,  /* 16bit default */
+			.dest_data_size = 1,    /* 16bit default */
+			.source_burst_length = 1,
+			.dest_burst_length = 0, /* SINGLE transfer */
+			.dma_callback = dma_tx_callback,
+		},
+		.stream_start = tx_stream_start,
+		.stream_disable = tx_stream_disable,
+		.queue_drop = tx_queue_drop,
+		.mem_block_queue.buf = tx_5_ring_buf,
+		.mem_block_queue.len = ARRAY_SIZE(tx_5_ring_buf),
+	},
+};
+DEVICE_AND_API_INIT(i2s_stm32_5, CONFIG_I2S_5_NAME, &i2s_stm32_initialize,
+		    &i2s_stm32_data_5, &i2s_stm32_config_5, POST_KERNEL,
+		    CONFIG_I2S_INIT_PRIORITY, &i2s_stm32_driver_api);
+
+static void i2s_stm32_irq_config_func_5(struct device *dev)
+{
+	SYS_LOG_DBG("Enable I2S5 IRQ");
+	IRQ_CONNECT(CONFIG_SPI_5_IRQ, CONFIG_SPI_5_IRQ_PRI, i2s_stm32_isr,
+		    DEVICE_GET(i2s_stm32_5), 0);
+	irq_enable(CONFIG_SPI_5_IRQ);
+}
+
+#endif /* CONFIG_I2S_5 */
+
+static inline struct device *get_dev_from_rx_dma_channel(u32_t dma_channel)
```

Comment:
That's a bit inconsistent. We generally don't place any functions after the device instantiation code. Let's move it somewhere above 
```
#ifdef CONFIG_I2S_1
```
line

Reply:
ack